src/components/Profile.svelte

```diff
@@ -1,17 +1,104 @@
 <script>
+  import { db } from "../utils/api";
   import { handleLogout } from "../utils/auth";
-  import { useNavigate, useLocation } from "svelte-navigator";
   import { user } from "../utils/stores";
+  import { getAuth, updatePassword, onAuthStateChanged } from "firebase/auth";
+  import {
+    onSnapshot,
+    collection,
+    deleteDoc,
+    doc,
+    updateDoc,
+  } from "firebase/firestore";
+  import { update } from "firebase/database";
+  import UpdateListings from "./UpdateListings.svelte";
+
+  const auth = getAuth();
+
+  let fireUser;
+
+  onAuthStateChanged(auth, (firebaseUser) => {
+    if (firebaseUser) {
+      fireUser = firebaseUser;
+    } else {
+      console.log("no user!");
+    }
+  });
 
   let signedIn;
 
   user.subscribe((value) => {
     signedIn = value;
   });
 
+  let newPassword = "";
+
+  function changePassword() {
+    console.log("inside function");
+    updatePassword(fireUser, newPassword)
+      .then(() => {
+        newPassword = "";
+        console.log("Password Changed");
```

Comment:
Good to give this message to the user instead of console log, perhaps using:  alert("Passwords Changed")

Reply:
Done!